src/changelog/linker/mapper/leadingv.go

```diff
@@ -0,0 +1,82 @@
+package mapper
+
+import (
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/Masterminds/semver"
+	"github.com/newrelic/release-toolkit/src/changelog"
+	"github.com/newrelic/release-toolkit/src/changelog/linker"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	checkTimeoutSeconds = 1
+)
+
+// LeadingVCheck wraps another mapper and performs a check over the result of Map. If the check is not
+// successful, it tries to generate it prepending a leading v to the version.
+type LeadingVCheck struct {
+	mapper    linker.Mapper
+	checkLink func(link string) bool
+}
+
+// WithLeadingVCheck returns a LeadingVCheck with the provided underlying mapper and a check function which
+// performs a request to the url corresponding to the link and check its status code.
+func WithLeadingVCheck(mapper linker.Mapper) *LeadingVCheck {
+	return &LeadingVCheck{mapper: mapper, checkLink: checkLinkResponse}
+}
+
+func (l *LeadingVCheck) Map(dep changelog.Dependency) string {
+	link := l.mapper.Map(dep)
+	if link == "" || l.checkLink(link) {
+		return link
+	}
```

Comment:
I'd suggest to split this `if` and add some `log.Trace` (or Debug) saying "Skipping link check as inner mapper did not return anything", and "Performing link check on %q".

Reply:
I included some logs here: [6f5cbde](https://github.com/newrelic/release-toolkit/pull/134/commits/6f5cbde1ffe7074c0185bed72543d9d8264ec1e3)